src/program/program.controller.ts

```diff
@@ -5,28 +5,44 @@ import { AuthGuard } from '~/auth/auth.guard';
 import { Local } from '~/decorator';
 import { ProgramService } from './program.service';
 
-@UseGuards(AuthGuard)
 @Controller({
   path: 'programs',
   version: '2',
 })
 export class ProgramController {
   constructor(private programService: ProgramService) {}
 
-  @Get()
-  async getProgramByMemberId(@Local('member') member: JwtMember, @Req() request: Request) {
-    const { memberId } = request.query;
-
-    return this.programService.getProgramByMemberId(member.appId, String(memberId || member.memberId));
-  }
-
+  @UseGuards(AuthGuard)
   @Get('/expired')
   async getExpiredProgramByMemberId(@Local('member') member: JwtMember, @Req() request: Request) {
     const { memberId } = request.query;
 
     return this.programService.getExpiredProgramByMemberId(member.appId, String(memberId || member.memberId));
   }
 
+  @UseGuards(AuthGuard)
+  @Get('/:programId')
+  async getProgramByMemberId(@Local('member') member: JwtMember, @Param('programId') programId: string) {
+    const { permissions } = member;
+
+    const extraAllowPermission = ['PROGRAM_NORMAL'].find((e) => permissions.includes(e));
+    return ['PROGRAM_ADMIN'].some((e) => permissions.includes(e))
+      ? this.programService.getProgramByProgramId(programId)
+      : this.programService.getProgramByMemberId(member.memberId, programId, extraAllowPermission);
+  }
+
+  @Get('/:programId/contents/:programContentId/trial')
```

Comment:
it seems the function did not use programId , maybe is not necessarily in the routes

Reply:
after discussion , maybe this is more restful design 
program/programId/contents/programContentId/trial

Reply:
for `/programs/:programId/contents/:programContentId/trial`
backend should verify:
1. ACL: programId -> programContentId
2. resource relationship: programId <> programContentId
ex. 
```
/programs/1/contents/1-1
/programs/1/contents/1-2

/programs/1/contents/2-1   # wrong
```

Reply:
app -> member(id, app_id) -> order(id, member_id) -> payment(id, order_id)
ex. 
demo -> 1 -> 1 -> 1
demo -> 1 -> 1 -> 2
demo -> 1 -> 2 -> 1
demo -> 1 -> 2 -> 2
demo -> 2 -> 1 -> 1
demo -> 2 -> 1 -> 2
demo -> 2 -> 2 -> 1
demo -> 2 -> 2 -> 2

app -> member  ,   app -> order -> payment
ex.
demo -> 1  ,  demo -> 1 -> 1
demo -> 1  ,  demo -> 1 -> 2
demo -> 1 ,   demo -> 2 -> 1
demo -> 1 ,   demo -> 2 -> 2
demo -> 2 ,   demo -> 3 -> 1
demo -> 2,    demo -> 3 -> 2
demo -> 2,    demo -> 4 -> 1
demo -> 2 ,   demo -> 4 -> 2